zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -512,6 +512,80 @@ impl ConfirmationsPolicy {
     pub fn allow_zero_conf_shielding(&self) -> bool {
         self.allow_zero_conf_shielding
     }
+
+    /// Returns the number of confirmations until a transaction output is considered spendable,
+    /// given information about the output and the inputs to the transaction that produced it.
+    ///
+    /// # Parameters
+    /// - `target_height`: The minimum height at which the output will be potentially spent.
+    /// - `pool_type`: The Zcash pool that the output was received into.
+    /// - `receiving_key_scope`: The ZIP 32 [`Scope`] of the key that received the output,
+    ///   or `None` if the scope is unknown or corresponds to the ephemeral transparent
+    ///   key scope.
+    /// - `mined_height`: The block height at which the transaction that produced the output was
+    ///   mined, if any.
+    /// - `tx_trusted`: A boolean flag indicating whether the received transaction has been
+    ///   explicitly marked as trusted by the user.
+    /// - `max_shielding_input_height`: For outputs that are the result of wallet-internal
+    ///   shielding transactions, the maximum height at which any transparent input to that
+    ///   transaction was received.
+    /// - `tx_shielding_inputs_trusted`: For outputs that are the result of wallet-internal
+    ///   shielding transactions, a flag indicating whether all transparent inputs to that
+    ///   transaction have been explicitly marked as trusted by the user.
+    #[allow(clippy::too_many_arguments)]
+    pub fn confirmations_until_spendable(
+        &self,
+        target_height: TargetHeight,
+        pool_type: PoolType,
+        receiving_key_scope: Option<Scope>,
+        mined_height: Option<BlockHeight>,
+        tx_trusted: bool,
+        max_shielding_input_height: Option<BlockHeight>,
+        tx_shielding_inputs_trusted: bool,
+    ) -> u32 {
+        let trusted_height = target_height.saturating_sub(u32::from(self.trusted));
+        let untrusted_height = target_height.saturating_sub(u32::from(self.untrusted));
+
+        match pool_type {
+            PoolType::Transparent => {
+                #[cfg(feature = "transparent-inputs")]
+                let zc_shielding = self.allow_zero_conf_shielding;
+                #[cfg(not(feature = "transparent-inputs"))]
+                let zc_shielding = false;
+
+                if zc_shielding {
+                    0
+                } else if receiving_key_scope == Some(Scope::Internal) {
+                    u32::from(self.trusted)
+                } else {
+                    u32::from(self.untrusted)
```

Comment:
These should be accurate for general spends; whether or not the wallet permits spending transparent directly is orthogonal to whether the `ConfirmationsPolicy` permits spending transparent at all.
```suggestion
                    confs_for_trusted
                } else {
                    confs_for_untrusted
```